pkg/controller/intrusiondetection/intrusiondetection_controller.go

```diff
@@ -420,7 +434,10 @@ func (r *ReconcileIntrusionDetection) Reconcile(ctx context.Context, request rec
 		rcertificatemanagement.CertificateManagement(&rcertificatemanagement.Config{
 			Namespace:       render.IntrusionDetectionNamespace,
 			ServiceAccounts: []string{render.IntrusionDetectionName},
-			TrustedBundle:   trustedBundle,
+			KeyPairOptions: []rcertificatemanagement.KeyPairOption{
+				rcertificatemanagement.NewKeyPairOption(intrusionDetectionCfg.ADAPIServerCertSecret, false, true),
```

Comment:
This should be:
```
rcertificatemanagement.NewKeyPairOption(intrusionDetectionCfg.ADAPIServerCertSecret, true, true),
```
The second param ensures that the secret is also created in the operator namespace. The controller responsible for creating the secret is the controller that should have this value set to true.

For context: A few lines down the value  is false for the typhaNodeTLS.NodeSecret. This is because the core_controller is the controller responsible for creating that secret.

Reply:
https://github.com/tigera/operator/pull/1931/commits/4546b11522d4ea1557bf00850577c3a07a6522aa